app/src/main/java/com/github/studydistractor/sdp/eventHistory/EventHistoryViewModel.kt

```diff
@@ -68,4 +83,50 @@ class EventHistoryViewModel(
         _uiState.update { EventHistoryUiState(usersEvent) }
     }
 
+    fun claimPoints(eventId: String): Task<Void> {
+        if (eventId.isEmpty()) return Tasks.forException(Exception("Empty eventId"))
+        for (cp in eventsClaimPoints){
+            if (cp.eventId == eventId){
+                if (cp.claimUser.contains(currentUser.id)){
+                    return Tasks.forException(Exception("Points already claimed"))
+                }
+
+                val event = getEvent(eventId) ?: return Tasks.forException(Exception("Null event"))
```

Comment:
This could be moved to the start of the function, after the first if statement, to avoid repetition.